generate.sh

```diff
@@ -207,60 +154,60 @@ then
 fi
 
 echo "-> Dockerfiles + devfiles" >&2
+{
 for ocp_ver in "${ocp_versions[@]}"
 do
-    # Soon this might be able to be simplified, as "binaryless" container, FROM scratch
+    pushd catalog/$ocp_ver
+    ocp_image_ver_name=
+    # OCP version 4.15 changed the image name
+    if [[ "$(printf '%s\n' "v4.14" "$ocp_ver" | sort -V | tail -n1)" == "v4.14" ]]
+    then
+        # Use old form <= v4.14
+        ocp_image_name="ose-operator-registry"
+    else
+        # Use new form > v4.14
+        ocp_image_name="ose-operator-registry-rhel9"
+    fi
+
+    opm generate dockerfile --binary-image registry.redhat.io/openshift4/$ocp_image_name:$ocp_ver $operator_name
+    mv $operator_name.Dockerfile ../Dockerfile
+
+    popd
+    
     {
-        cat <<EOF
-# The base image is expected to contain
-# /bin/opm (with a serve subcommand) and /bin/grpc_health_probe
-EOF
-
-        # OCP version 4.15 changed the image name
-        if [[ "$(printf '%s\n' "v4.14" "$ocp_ver" | sort -V | tail -n1)" == "v4.14" ]]
-        then
-            # Use old form <= v4.14
-            echo "FROM registry.redhat.io/openshift4/ose-operator-registry:$ocp_ver"
-        else
-            # Use new form > v4.14
-            echo "FROM registry.redhat.io/openshift4/ose-operator-registry-rhel9:$ocp_ver"
-        fi
-
-        cat <<EOF
-
-# Configure the entrypoint and command
-ENTRYPOINT ["/bin/opm"]
-CMD ["serve", "/configs", "--cache-dir=/tmp/cache"]
-
-# Copy declarative config root into image at /configs and pre-populate serve cache
-ADD $operator_name /configs/$operator_name
-RUN ["/bin/opm", "serve", "/configs", "--cache-dir=/tmp/cache", "--cache-only"]
-
-# Set DC-specific label for the location of the DC root directory
-# in the image
-LABEL operators.operatorframework.io.index.configs.v1=/configs
+    cat <<EOF
```

Comment:
Here's the replacement devfile generation, instead of using the composite template.

Reply:
Even more so than the Dockerfile, there are some differences in the output now. The `kubernetes:` line seems like a syntax error now? Or is that part of intended changes?

```diff
diff --git a/catalog/v4.11/devfile.yaml b/catalog/v4.11/devfile.yaml
index b7043d4..6897155 100644
--- a/catalog/v4.11/devfile.yaml
+++ b/catalog/v4.11/devfile.yaml
@@ -1,7 +1,7 @@
 schemaVersion: 2.2.0
 metadata:
-  name: fbc-v4.11
-  displayName: FBC v4.11
+  name: fbc-rhbk-operator
+  displayName: FBC rhbk-operator
   description: 'File based catalog'
   language: fbc
   provider: Red Hat
@@ -10,17 +10,17 @@ components:
     image:
       imageName: ""
       dockerfile:
-        uri: Dockerfile
+        uri: .Dockerfile
         buildContext: ""
-  - name: kubernetes
-    kubernetes:
-      inlined: placeholder
+  - kubernetes:
+    inlined: "placeholder"
     attributes:
-      deployment/container-port: 50051
-      deployment/cpuRequest: 100m
-      deployment/memoryRequest: 512Mi
-      deployment/replicas: 1
-      deployment/storageRequest: "0"
+    - deployment:
+        container-port: 50051
+        cpuRequest: "100m"
+        memoryRequest: "512Mi"
+        replicas: 1
+        storageRequest: 0
 commands:
   - id: build-image
     apply:
```

Reply:
Dropping the devfile stuff, since evidently it's no longer needed.